### ISSUE TITLE
Password doesn't work on latest release

### ISSUE THREAD

**Issue by knightjdr** [turtlecoin/desktop-xamarin#20]
I can no longer open my wallet with the current release 1.0.6625.36022.

I enter the password and the daemon starts but then an alert opens with "incorrect password". It was working in the previous release.

**Comment by ZedPea** [comment]
I saw a similar issue on an unsynced daemon. Upon creating a new wallet, the "The password is incorrect" message popped up quite a few times whilst in the background the wallet was syncing with the chain. Once it did sync fully it was fine.

**Comment by knightjdr** [comment]
It does sync in the background for me, but once synced it still says incorrect password and if I click the alert it just comes back again.

**Comment by codingwithredbull** [comment]
This is most likely because the password bug was fixed, and you unknowingly have a different password than you think.

Let me give you an example. There was a bug in the last versions where special characters in passwords would cause the password to mess up and the password would be only the characters before the first special character. This is because the wallet is created through command line with walletd as there is no other options. So when you create a wallet its essentially the same as opening a cmd prompt and typing:
`walletd.exe -g -w main.wallet -p password`

So now lets take a example of how this would happen with a persons password. Lets say that a person wanted to make a wallet named `Main` and a password of `MyStRoNg"PaSsWoRd`. This command would be similar to the follow:
`walletd.exe -g -w Main.wallet -p MyStRoNg\PaSsWoRd`
and that is exactly what the old versions would do. However, look at that special character there. Because of the command escaping, this tims off `"PaSsWoRd` cause the `\` and now the password really only is `MyStRoNg`. Now when the user goes to open their wallet, it uses the same `MyStRoNg"PaSsWoRd` password, but again the end is trimmed, so the user never thinks their password is a problem and the world continues as normal.

However, with the last update, this was fixed, and proper escaping occurs now. So when the gui goes to open your wallet it now uses:
`walletd.exe -g -w Main.wallet -p "MyStRoNg\PaSsWoRd"`
Notice the quotes? This means the password is now whole as it was intended to be, but your wallets password is actually still `MyStRoNg`, causing the incorrect password prompt. 

Sorry for the issues, sadly this went unnoticed for a very long time. My suggestion would be to try truncated versions of your password until opening it is successful. At that point i would create a new wallet and send your funds to this new wallet if the truncated password is a problem.

**Comment by knightjdr** [comment]
It's no problem. I have simplewallet running without problems so there isn't a real issue with accessing my wallet itself.

Because of the previous password issues the password I created was 15 characters only letters and numbers, so no special characters. It's stored in a password manager so just copy pasting. Maybe it's just something particular to me. Feel free to close this issue if no one else comes along.

**Comment by codingwithredbull** [comment]
Hmmm, this is interesting. Could you please download a process monitoring app like "[ProcessHacker](https://processhacker.sourceforge.io/)" or "[ProcMon](https://docs.microsoft.com/en-us/sysinternals/downloads/procmon)" and view the `Walletd.exe` process and see if the password in the commandline matches the password you know?

It may reveal what is going on.

**Comment by Turtle2018** [comment]
I downloaded the TurtleWallet_1.0.6625.36022.zip from github and synced the block till date but when I close the wallet and open it keep on asking for password. The popup appears as " In correct Password" several times.when I closed it the attached error appeared. it worked fine for first time, but when i close it I cannot use the wallet anymore. 

I ended up cleaning  the target directory and extracted the TurtleWallet_1.0.6625.36022.zip . Repeated this steps several times but no success.
[turtle_wal_error.txt](https://github.com/turtlecoin/desktop-xamarin/files/1749995/turtle_wal_error.txt)
![turtle_error](https://user-images.githubusercontent.com/36749191/36571205-2a33ed12-17fb-11e8-9f1d-7ac53c72080e.PNG)

Operating system: Windows 10

**Comment by ngaustad** [comment]
I'm having the same issue, so I created a new wallet, and I'm still getting password incorrect.

**Comment by codingwithredbull** [comment]
@Turtle2018  and @ngaustad  can you please follow the instructions here and post back to me? it would really help:

Hmmm, this is interesting. Could you please download a process monitoring app like "[ProcessHacker](https://processhacker.sourceforge.io/)" or "[ProcMon](https://docs.microsoft.com/en-us/sysinternals/downloads/procmon)" and view the `Walletd.exe` process and see if the password in the commandline matches the password you know?

It may reveal what is going on.

**Comment by ngaustad** [comment]
@codingwithredbull  Yes the command line does show the correct password.

**Comment by codingwithredbull** [comment]
Wow that is very unexpected. Could you please delete walletd.log file, and run through the motions again only once of attempting to open the wallet, get the error prompt, and post the walletd.log file here for me? Thanks for the cooperation!  @ngaustad

**Comment by ngaustad** [comment]
Here you go:
[walletd.log](https://github.com/turtlecoin/desktop-xamarin/files/1750066/walletd.log)

**Comment by ngaustad** [comment]
Hum... it seems after deleting the log file I was able to get into the test wallet I created... still can't get into my main wallet

**Comment by ngaustad** [comment]
I'm fairly certain I was able to sign in right after the update so I don't think my password changed like you suggested earlier, but just to be sure did that bug exist in the previous GUI version?

**Comment by knightjdr** [comment]
After deleting the walletd.log I was able to access my wallet without problems. I tried closing and signing in again and it still works. From inspecting the log it looks like even when it was telling me my password was incorrect it had opened my wallet. So seems to have been purely a gui issue for me.

**Comment by ngaustad** [comment]
@codingwithredbull however, say it did, if the "new" password is shorter than the minimum character limit, what may happen?

**Comment by ngaustad** [comment]
@codingwithredbull Ok I found another issue it seems that if you create a password with a space " " it will always tell you the password is incorrect.

**Comment by higginsgy** [comment]
@codingwithredbull 
Hello, i have two wallets - 
1) use only numbers and symbols in password
2) use double _ and ! in password 

I can no longer open my second wallet (first wallet - ok) with the current release 1.0.6625.36022.

**Comment by Bcdo** [comment]
I got this issue when I first tried to create a wallet, but when I created a new wallet with only letters and numbers, and not with space and symbols as I usually use, then it worked.

**Comment by brador2000** [comment]
I figured this out - for me at least. Perhaps it is the same for some of you.
When creating a wallet in the latest version I was able to to use a 26 character password and the wallet created just fine, BUT then when I close the wallet and go to reopen it in that very same version I get the wrong password error. It is logged as such as well. I checked the passwordPrompt.Designer.cs file for 1.0.6625.36022 Update and sure enough on line 149 it says `this.passwordText.MaxLength = 24;` 
Another file - passwordPrompt.cs - looks like it only checks for the password on a newly generated wallet to be greater than six characters but has no upper limit, as per line 30 `if (passwordText.Text != "" && passwordText.Text.Length > 6)` so that's why I was able to use my 26 character password to create the wallet but not to open it. I had to downgrade to 1.0.6617.7635 so that I could open the wallet I created with 1.0.6625.36022

**Comment by ZedPea** [comment]
@brador2000 I've made a PR which addresses this, it ups the limit to 150. However, I didn't put a check on the max length, as I'm not sure it's possible, since the length will always be capped at 150, you can't check that the password is greater than that length, as any extra characters will be truncated.

I suppose you could add a onTextChanged event and alert them when their password reaches the limit instead, I'll look into this.

This patch should also fix any wallet passwords being encoded incorrectly on the command line, but will potentially cause some people's wallets to no longer open if their passwords were being silently truncated. 

Meanwhile, if you want to change your password - export your keys, then import them into simplewallet and give that wallet a working password. Reopen this new wallet in the GUI. I'm going to look into adding an import keys option on the GUI as well to streamline the process.

**Comment by knightjdr** [comment]
In relation to all this, it might be good to force or at least remind people to export their keys before doing an automatic update in case password issues continue to be a problem. Just an alert that opens from which they can either initiate the export before the update or dismiss it if they already have an export.

**Comment by brador2000** [comment]
@ZedPea Thanks for the info. I am new to GitHub commenting even though I've lurked for years and should have realized to check the PRs first.

**Comment by ZedPea** [comment]
@CrypOps Password encoding has been fixed, and so some passwords which were previously encoded incorrectly may now no longer work. If you have your private keys, use them to import, otherwise, download the previous release, export your keys, and then use them to import.

**Comment by CrypOps** [comment]
@ZedPea When trying to Backup Keys it's saying Incorrect Password!

I am using TurtleWallet_1.0.6625.36022.

When launching the wallet the password works only for the v1.0.6625.36022, and when I send coins it says the transaction is successful but the coins are not actually being sent, when I reopen the wallet it displays the same balance as before.
